SEPA DD R-Flows: Returns (#391)
clearbank/clearbank.github.io: content/eu/direct/sepa-direct-debit.mdx

]
```

## SEPA Direct Debit R-flows

### Overview

> SEPA Direct Debit has three 'R-flows': Refuse, Return, and Refund. Each flow represents an outcome in the SEPA Direct Debit payment flow. The safest way to reconcile R-flows is by using canonical payment identifiers.

Reviewer: ```suggestion
SEPA Direct Debit has three 'R-flows': Refuse, Return, and Refund. Each R-flow represents an outcome in the payment lifecycle. The safest way to reconcile R-flows is by using canonical payment identifiers.
```

### Refuse

> _Details TBC_

Reviewer: Make this into a callout?

### Return

> A return is a request submitted by the client after a SEPA Direct Debit has settled, instructing the bank to return the payment to the debtor within the scheme’s permitted timeframe (up to 5 working days after settlement).

Reviewer: ```suggestion
A return is a request submitted by the client after a SEPA Direct Debit has settled, instructing the bank to return the funds back to the debtor. This must be requested within the scheme’s permitted timeframe (up to 5 working days after settlement).
```

> A return can be submitted when a client determines that a settled SEPA Direct Debit should be returned to the debtor, for example due to insufficient funds or account restrictions. The client must specify a valid return reason code when submitting the request.

Reviewer: Who is 'client' in this scenario? Our customer? Or the end user?

Returns may be requested and received up to 5 working days after the date of settlement.

Returns have standard reason codes to help you resolve and communicate the underlying reason for a return:

#### Return reason code table

| Reason Code | Description                         |
| ----------- | ----------------------------------- |
| `AC06`      | BlockedAccount                      |
| `AM04`      | InsufficientFunds                   |
| `BE05`      | UnrecognisedInitiatingParty         |
| `MD01`      | NoMandate                           |
| `MD07`      | EndCustomerDeceased                 |
| `MS02`      | NotSpecifiedReasonCustomerGenerated |
| `MS03`      | NotSpecifiedReasonAgentGenerated    |
| `RR04`      | RegulatoryReason                    |

#### Simulate a SEPA DD Return

> To simulate a SEPA DD return, call our [POST /payments/sepa-dd/v1/r-message-instructions](#simulate-an-r-flow-for-a-sepa-direct-debit-payment) endpoint with an appropriate return reason code and reference to the original payment set out in [POST /payments/sepa-dd/v1/emulator/inbound-direct-debits](#simulate-an-inbound-sdd-payment-request).

Reviewer: ```suggestion
To simulate a SEPA DD return, call the [POST /payments/sepa-dd/v1/r-message-instructions](#simulate-an-r-flow-for-a-sepa-direct-debit-payment) endpoint. Fields must be populated with an appropriate return reason code and the original payment ID set out in [POST /payments/sepa-dd/v1/emulator/inbound-direct-debits](#simulate-an-inbound-sdd-payment-request).
```

Reviewer: It says "original payment ID" in the API, but original payment reference here. Which is it?
